src/encryption.rs

```diff
@@ -116,5 +116,32 @@ mod test {
     let decrypted_data = decrypt(&encrypted_data[..], &key, &iv).ok().unwrap();
 
     assert!(message.as_bytes() == &decrypted_data[..]);
-  } 
+  }
```

Comment:
This code should be in Self Encryption (lib.rs) and decorate the existing struct. So if you can move it there and update tests etc. then we are good to go.